main.cpp

```diff
@@ -41,23 +41,310 @@ If you need inspiration for what to write, take a look at previously approved st
 /*
  UDT 1:
  */
+#include <iostream>
+
+struct Animal
+{
+    int totalCalories;
+    bool isHungry = true;
+    int mealsEaten;
+    float distanceWalked;    
+    float animalHeight{ 8.2f };
+
+    Animal();
+    ~Animal();
+    void walk( float currentPace, int timeWalked, bool walkForward);
+    void grow( float growthHeight = 0.2f );
+    void consume( int calorieAmount = 1600 );
+
+};
+
+Animal::Animal() : 
+totalCalories( 2600 ),
+mealsEaten(3),
+distanceWalked( 150.5f ) {}
+
+Animal::~Animal()
+{
+    std::cout << "Animal Destroyed" << std::endl;
+}
+
+void Animal::walk(float currentPace, int timeWalked, bool walkForward)
+{
+   if( walkForward ){
+       distanceWalked += (currentPace * timeWalked); 
+   } else{
+       distanceWalked -= (currentPace * timeWalked);
+   }
+}
+
+void Animal::grow(float growthHeight)
+{
+    for(int i = 0; i < 6; ++i )
+    {
+        animalHeight += growthHeight;
+    }
+}
+
+void Animal::consume( int calorieIntake )
+{
+    if(isHungry){
+        totalCalories += calorieIntake;
+        mealsEaten++;
+        isHungry = false;
+    }
+}
 
 /*
  UDT 2:
  */
+struct Ball
+{
+    double ballRadius;
+    float ballWeight;
+    bool canRoll = true;
+    bool isBouncy = false;
+    int numberOfDimples = 201;
+
+    struct Sandball
+    {
+        float percentFilled = 75.0f;
+        bool plasticCoating = false;
+        float sandWeight;
+        bool fineSandFilling;
+        int lengthOfTail;
+
+        Sandball();
+        ~Sandball();
+        void fillBall(int howLong);
+        void changeFilling();
+        void cutTail(int cutAmount);
+    };
+
+    Ball();
+    ~Ball();
+    void getHeavier(bool soakInWater, int soakingTime = 3);
+    double getVolume(bool isHollow);
+    void addDimples(int howManyToAdd);
+};
+
+Ball::Ball() : 
+ballRadius(2.5),
+ballWeight(4.2f){}
+
+Ball::~Ball()
+{
+    std::cout << "Ball Destroyed" << std::endl;
+}
+
+void Ball::getHeavier(bool soakInWater, int soakingTime)
+{
+    if(soakInWater)
+    {
+        while(soakingTime > 0)
+        {
+            //std::cout << "Soaked Ball" << std::endl;
+            --soakingTime;
+            ballWeight += 0.2f;
+        }
+    }
+}
+
+double Ball::getVolume(bool isHollow)
+{
+    if(isHollow)
+    {
+        return 0.0;
+    }
+    double currentVolume;
+    currentVolume = (4/3)*(3.14159)*(ballRadius * ballRadius);
+    return currentVolume;
```

Comment:
just return line 160.  no need for this separate declaration/assignment/return.
- [ ] revise.